### DISCUSSION TITLE
Hi guys! I have a problem with FAST LED library.

### DISCUSSION THREAD

**Discussion by V2mD** [dbuezas/lgt8fx#199] [Q&A] [answered]
Hi, guys, i bought the NANO PRO version board with LGT8F - 328p. So, i started blink with this board with next parameters: board arduuino Nano, processor - atmega328p  and with  AVR ISP as programmator. Blink works correct. But if i load any sketch with FastLED, i get diode with max bright of white color. I tryed to configure board with yours core - install board and choose nano version, but get this:
Arduino: 1.8.13 (Windows 10), Плата:"Arduino Nano, ATmega328P"

Скетч использует 21952 байт (73%) памяти устройства. Всего доступно 29696 байт.

Глобальные переменные используют 1121 байт динамической памяти.

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 3 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 4 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 5 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 6 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 7 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 8 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 9 of 10: not in sync: resp=0xe6

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 10 of 10: not in sync: resp=0xe6

Проблема загрузки в плату. Помощь по загрузке: http://www.arduino.cc/en/Guide/Troubleshooting#upload .

Этот отчёт будет иметь больше информации с
включенной опцией Файл -> Настройки ->
"Показать подробный вывод во время компиляции"
Thank you for attention!

**Comment by ithinkido** [comment]
This does not look right 
`Arduino: 1.8.13 (Windows 10), Плата:"Arduino Nano, ATmega328P`
That should be LGT8F..... Are you sure you have selected the correct board?
![image](https://user-images.githubusercontent.com/52834821/154061718-dd1739a1-66b3-4580-8950-65f794b20285.png)

**Reply by jayzakk** [reply]
He did not, he even wrote that he selected atmega328p, not lgtf :)

**Reply by V2mD** [reply]
![image](https://user-images.githubusercontent.com/99718575/154069367-18408ce6-dcd3-4660-b449-6912703afe59.png)

**Reply by V2mD** [reply]
Arduino: 1.8.13 (Windows 10), Плата:"LGT8F328, Default (64), Internal, 32 MHz, 328P-LQFP48 MiniEVB"

Скетч использует 21902 байт (73%) памяти устройства. Всего доступно 29696 байт.

Глобальные переменные используют 935 байт динамической памяти.

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00

Проблема загрузки в плату. Помощь по загрузке: http://www.arduino.cc/en/Guide/Troubleshooting#upload .

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 3 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 4 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 5 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 6 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 7 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 8 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 9 of 10: not in sync: resp=0x00

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 10 of 10: not in sync: resp=0x00



Этот отчёт будет иметь больше информации с
включенной опцией Файл -> Настройки ->
"Показать подробный вывод во время компиляции"

**Reply by V2mD** [reply]
Sorry, i replyed right moment now)

**Reply by ithinkido** [reply]
Do you have a photo of your board?

**Reply by V2mD** [reply]
![WhatsApp Image 2022-02-15 at 18 22 06](https://user-images.githubusercontent.com/99718575/154070625-43387d7a-c56f-4a19-9fbb-1e73958dff39.jpeg)
it nano pro

**Reply by ithinkido** [reply]
Presuming that is a LGT8F328p chip on board, you still have the wrong board definition.
`LGT8F328, Default (64), Internal, 32 MHz, 328P-LQFP48 MiniEVB`
should be 
LGT8F328, Default (64), Internal, 32 MHz, 328P-LQFP***32*** (e.g. MiniEVB nano-style and WAVGAT)

Try that first. 
Do you get the same errors

```
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x00
....
....
....

```

when you upload a normal blink sketch ?

**Reply by V2mD** [reply]
yep. If i choosing the  arduino board and 328p - it's works

**Reply by V2mD** [reply]
Arduino: 1.8.13 (Windows 10), Плата:"LGT8F328, Default (64), Internal, 32 MHz, 328P-LQFP32 (e.g. MiniEVB nano-style and WAVGAT)"

Скетч использует 4844 байт (16%) памяти устройства. Всего доступно 29696 байт.

Глобальные переменные используют 100 байт динамической памяти.

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 3 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 4 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 5 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 6 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 7 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 8 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 9 of 10: not in sync: resp=0xa2

avrdude: stk500_recv(): programmer is not responding

avrdude: stk500_getsync() attempt 10 of 10: not in sync: resp=0xa2

Проблема загрузки в плату. Помощь по загрузке: http://www.arduino.cc/en/Guide/Troubleshooting#upload .



Этот отчёт будет иметь больше информации с
включенной опцией Файл -> Настройки ->
"Показать подробный вывод во время компиляции"

**Reply by ithinkido** [reply]
> yep. If i choosing the arduino board and 328p - it's works

To me it sounds like you have an ATmega328p chip then, and not a LGT8F328p chip.

It is not possible to upload code to a chip that is different from the chip selected in Arduino IDE. In my case, I have a LGT8F328p chip . If I select ATmeg328p then I get the same error as you

![image](https://user-images.githubusercontent.com/52834821/154078009-648160d2-be5b-481b-a653-956ab783f265.png)

**Reply by LaZsolt** [reply]
@V2mD 
Your board is a very new design. It's upload speed is 115200 bps. 
dbuzeas/lgt8fx package default upload speed is 57600 bps.

The solution: Replace the upload speed in line 18 in file `C:\Users\(yourusernae)\AppData\Local\Arduino15\packages\LGT8fx Boards\hardware\avr\1.0.6\boards.txt`

**Reply by V2mD** [reply]
YES!!!! WORKS!!!! THANK YOU!   YOU ARE THE BEST!

**Reply by V2mD** [reply]
But, if i want to change board to older version, i will must return previous value?

**Reply by LaZsolt** [reply]
(The link of boards.txt is edited.)
Here a new `boards.txt` :
https://github.com/dbuezas/lgt8fx/blob/fb4f06a097e6a225a20d1b78d7e70eac160a6d5b/lgt8f/boards.txt
1. Close Arduino IDE.
2. You have to rename your existing `boards.txt` for save the original file.
3. Copy to your HDD drive the new boards.txt.
4. Start Arduino IDE.
Finally you will find two custom boards settings with 16 or 12 MHz crystal. If you using the internal oscillator then the external crystal does not matter, but you could select the upload speed.

**Reply by V2mD** [reply]
If i used new file "Boards"  i see new variants, but i get compilation fault
![image](https://user-images.githubusercontent.com/99718575/154260609-a69d4dad-38af-45d8-ad1c-c1f28f8610e2.png)
![image](https://user-images.githubusercontent.com/99718575/154260975-d4efb469-64db-4be5-bf17-bebdc58b0c13.png)
![image](https://user-images.githubusercontent.com/99718575/154261059-08045b74-7a11-4d27-bcfc-a4ee64f12890.png)
Arduino: 1.8.13 (Windows 10), Плата:"LGT8F328 + 12 MHz crystal, Default (64), Internal 32 MHz (Not for 328D), 328P-LQFP32, 115200"

In file included from C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/platforms/avr/fastled_avr.h:6:0,

                 from C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/platforms.h:41,

                 from C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/FastLED.h:54,

                 from C:\Users\V2m\Desktop\WS2812_FX-master\прошивки\WS2812_FX_rndChange_light_fixed\WS2812_FX_rndChange_light_fixed.ino:8:

C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/platforms/avr/clockless_trinket.h:19:11: error: operator '==' has no left operand

 #if (F_CPU==8000000)

           ^~

In file included from C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/FastLED.h:67:0,

                 from C:\Users\V2m\Desktop\WS2812_FX-master\прошивки\WS2812_FX_rndChange_light_fixed\WS2812_FX_rndChange_light_fixed.ino:8:

C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/fastspi.h:13:40: error: operator '>' has no left operand

 #if defined(FASTLED_TEENSY3) && (F_CPU > 48000000)

                                        ^

In file included from C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/FastLED.h:68:0,

                 from C:\Users\V2m\Desktop\WS2812_FX-master\прошивки\WS2812_FX_rndChange_light_fixed\WS2812_FX_rndChange_light_fixed.ino:8:

C:\Users\V2m\Documents\Arduino\libraries\FastLED\src/chipsets.h:457:48: error: operator '==' has no left operand

 #if defined(__LGT8F__) || (CLOCKLESS_FREQUENCY == 8000000 || CLOCKLESS_FREQUENCY == 16000000 || CLOCKLESS_FREQUENCY == 24000000) //  || CLOCKLESS_FREQUENCY == 48000000 || CLOCKLESS_FREQUENCY == 96000000) // 125ns/clock

                                                ^~

Несколько библиотек найдено для "FastLED.h"

 Используется: C:\Users\V2m\Documents\Arduino\libraries\FastLED

Не используется: D:\arduino-1.8.13\libraries\FastLED-stm32patch

exit status 1

Ошибка компиляции для платы LGT8F328 + 12 MHz crystal.



Этот отчёт будет иметь больше информации с
включенной опцией Файл -> Настройки ->
"Показать подробный вывод во время компиляции"

**Reply by V2mD** [reply]
and if i used the first variant - all works correctly

**Reply by LaZsolt** [reply]
You need to replace two more files:
https://github.com/dbuezas/lgt8fx/blob/fb4f06a097e6a225a20d1b78d7e70eac160a6d5b/lgt8f/cores/lgt8f/main.cpp
https://github.com/dbuezas/lgt8fx/blob/fb4f06a097e6a225a20d1b78d7e70eac160a6d5b/lgt8f/platform.txt

I checked the Fastled and found, the library can't run on 32 MHz.

**Reply by V2mD** [reply]
So i must use external resonator?

**Reply by LaZsolt** [reply]
No, you don't need to use external clock source.
You can select lower core frequencies in the Arduino IDE when the internal clock source selected.

**Reply by V2mD** [reply]
Thank You, Dear LaZsolt, i changed these files, and compilation works with new board

**Reply by LaZsolt** [reply]
Your welcome.

**Comment by LaZsolt** [comment, marked as answer]
It is expected that there will be difficulties with LED control, because of in the current release v1.0.6 the function `delayMicroseconds()` delaying about 25% shorter.

**Reply by LaZsolt** [reply]
The Fastled library may not be using the `delayMicroseconds()` function.
